components/org.wso2.carbon.healthcheck.api.core/src/main/java/org/wso2/carbon/healthcheck/api/core/CarbonMemoryAgentService.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.healthcheck.api.core;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.healthcheck.api.core.util.HealthCheckConfigParser;
+import sun.management.HotspotInternal;
+import sun.management.HotspotThreadMBean;
+
+import javax.management.*;
+import java.lang.management.*;
+import java.util.EnumMap;
+import java.util.Timer;
+import java.util.TimerTask;
+
+/**
+ * Service which logs the memory usage of carbon server.
+ */
+public class CarbonMemoryAgentService {
+
+    static Timer timer = new Timer("MemoryAgent Timer", true);
+    static int BYTES_PER_MB = 1024 * 1000;
+    public static final String INTERNAL_MBEAN = "wso2is:name=hostpotInternal";
+    public static final String THREADING_MBEAN = "sun.management:type=HotspotThreading";
+    private static MemoryMXBean memoryMXBean;
+    private static ThreadMXBean threadMXBean;
+    private static MBeanServer mBeanServer;
+    private static HotspotThreadMBean hotspotThreadMBean;
+
+    private static final Log log = LogFactory.getLog(CarbonMemoryAgentService.class);
+    private static CarbonMemoryAgentService instance = new CarbonMemoryAgentService();
+
+    private CarbonMemoryAgentService() {
+
+    }
+
+    public static CarbonMemoryAgentService getInstance() {
+
+        memoryMXBean = ManagementFactory.getMemoryMXBean();
+        threadMXBean = ManagementFactory.getThreadMXBean();
+        mBeanServer = ManagementFactory.getPlatformMBeanServer();
+        try {
+            mBeanServer.registerMBean(new HotspotInternal(), objectName(INTERNAL_MBEAN));
+        } catch (InstanceAlreadyExistsException | MBeanRegistrationException | NotCompliantMBeanException |
+                MalformedObjectNameException e) {
+            log.error("Failed to register object as an MBean.", e);
```

Comment:
I think in case if required mbeans are not available, there is no point in giving an instance outside. Hence we shouldn't handle this exception.

Reply:
Ideally I think these initialisations should be done in the constructor.

Reply:
removed the exception handling and mbean registering. I have used the getInstance method for initializations since we normally make OSGi components singleton